tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -2212,3 +2214,67 @@ def benford_correlation(x):
     # np.corrcoef outputs the normalized covariance (correlation) between benford_distribution and data_distribution.
     # In this case returns a 2x2 matrix, the  [0, 1] and [1, 1] are the values between the two arrays
     return np.corrcoef(benford_distribution, data_distribution)[0, 1]
+
+
+@set_property("fctype", "combiner")
+def matrix_profile(x, param):
+    """
+    TODO: Documentation
+
+    :param x: the time series to calculate the feature of
+    :type x: numpy.ndarray
+    :param param: contains dictionaries {"sample_pct": x, "threshold": y, "feature": z}
+                  with sample_pct and threshold being parameters of the matrixprofile
+                  package https://matrixprofile.docs.matrixprofile.org/api.html#matrixprofile-compute
+                  and feature being one of "min", "max", "mean", "median", "25", "75"
+                  and decides which feature of the matrix profile to extract
+    :type param: list
+    :return: the different feature values
+    :return type: pandas.Series
+    """
+    x = np.asarray(x)
+
+    def _calculate_pmp(**kwargs):
+        """Calculate the pan-matrix profile"""
+        try:
+            pmp = mp.compute(x, **kwargs)["pmp"]
+            # TODO: reasonable?
+            return pmp[(~np.isnan(pmp)) & (~np.isinf(pmp))]
+        except Exception:
+            return [np.NaN]
+
+    # The already calculated matrix profiles
+    matrix_profiles = {}
+
+    # The results
+    res = {}
+
+    for kwargs in param:
+        key = convert_to_output_format(kwargs)
+        feature = kwargs.pop('feature')
+
+        # Only calculate the pmp if we have not already done so
+        # The feature calculation can happen afterwards
+        featureless_key = convert_to_output_format(kwargs)
+        if featureless_key not in matrix_profiles:
+            matrix_profiles[featureless_key] = _calculate_pmp(**kwargs)
+
+        pmp = matrix_profiles[featureless_key]
+
+        # TODO: is this what we want?
```

Comment:
Another way we could calculate Tukey's Five Numbers is as follows (h/t to @tylerwmarrs):

```
  if data.size < 1:
      return [np.nan] * 5
  
  x = data[~np.isnan(data)]
  x = np.sort(x)
  n = len(x)
  n4 = np.floor((n + 3) / 2) / 2
  d = np.array([1, n4, (n + 1) / 2, n + 1 - n4, n])
  i1 = (np.floor(d) - 1).astype(int)
  i2 = (np.ceil(d) - 1).astype(int)
  
  return 0.5 * (x[i1] + x[i2])
```

The above way may not actually be faster/easier given that Numpy has already optimized the percentile count, so I'm fine if we leave as-is for now.

Reply:
So as I do not think that this part will be the performance bottleneck and the other code (at least in my opinion) is easier to read, I would go with the already present one (as you have mentioned)